src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObject = {};
+  const sortDate = [];
+  const [, , , separatorFrom] = fromFormat;
+  const [, , , separatorTo] = toFormat;
+  const separateDate = date.split(separatorFrom);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY') {
+      if (separateDate[i] < 30) {
+        dateObject[fromFormat[i] + 'YY'] = '20' + separateDate[i];
+      } else {
+        dateObject[fromFormat[i] + 'YY'] = '19' + separateDate[i];
+      }
+    } else if (fromFormat[i] === 'YYYY' && toFormat.includes('YY')) {
+      dateObject['YY'] = separateDate[i].slice(2);
```

Comment:
When slicing the year to convert from 'YYYY' to 'YY', you should check if `toFormat` at the current index is actually 'YY'. Right now, the check `toFormat.includes('YY')` is not index-specific and could lead to incorrect conversions if 'YY' is not at the same position in `toFormat` as 'YYYY' is in `fromFormat`.